### ISSUE TITLE
Dealing with Foes

### ISSUE THREAD

**Issue by heinrich5991** [ddnet/ddnet#5354]
converted from #4525.

@Scrumplex wrote:
> DDNet offers a foe list which will block communication from any player in that list.
>
> Currently there is no visual indicator and not even a UI to add players to the foe list. You can only add foes using the in-game console (probably F1 for you) with the command add_foe "<player name>" "<clan tag>".
>
> I propose a few things:
>
> * Adding / Removing foes with the UI, similar to the friends feature
>   * Being able to add/remove players on the same server
>   * Being able to add/remove players from the server browser's player list
> * Making foes and friends the mutually exclusive
>   * Friends always take priority
>   * Foes can't be added if they are friends
>   * If foes are made friends, they will be removed from foe list
> * Somehow showing that a player is a foe in-game.
>   * Redact the player name and replace with something else
>     * We could adopt CS:GO's way, by changing the name to a color
>   * Maybe render their name in italics so that the player will be able to tell if someone is foe-ed
> * Exposing more configuration options to the player on how to deal with
>    * Maybe the player doesn't want names redacted?
>    * Or they don't want so see emotes from foes?
>    * Maybe also force the default skin on them?
>
> Ongoing efforts:
> #4181

**Comment by kaitlynia** [comment]
I don't agree with "Friends always take priority" and "Foes can't be added if they are friends" since I think they would override each other bidirectionally. I don't know how useful changing the foe names would be; you likely already know if someone is foed based on their name. I think a more complete list is needed to address the last bullet, but I'm still unsure how useful any of that would be. Does anyone personally foe a lot of users, and can any of those people provide a list of things they actually want?
